### PR TITLE
Create destroy_runner method

### LINKED ISSUE #85: Run Runner Cleanup functions
The CL/AJAX runners that are initialized in the `Plugin_Request_Utility` have a prepare method that returns a cleanup function that needs to be run at the end of the process.

[See this comment for details](https://github.com/10up/plugin-check/pull/80/files#r1110168672).

## Acceptance Criteria
- Create a static `destroy_runner` method for the `Plugin_Request_Utility`.
-  This method will run cleanup function returned by the runners `prepare()` method and stored in the `static::$cleanup` variable.
- After calling the cleanup function, the `Plugin_Request_Utility::$runner` variable is set to `null`

### PR DESCRIPTION
**Pull request by jjgrainger**
Addresses acceptance criteria in #85 

Closes #85

### CODE REVIEW

File: includes/Utilities/Plugin_Request_Utility.php [hunk 1]
```diff
@@ -114,4 +114,15 @@ public static function get_runner() {
...
 	}
+
+	/**
+	 * Runs the cleanup functions and destroys the runner.
+	 *
+	 * @since n.e.x.t
+	 */
+	public static function destroy_runner() {
+		// Run the cleanup functions.
+		call_user_func( self::$cleanup );
```
Comment (felixarntz) [inline includes/Utilities/Plugin_Request_Utility.php]: We need to ensure that `self::$cleanup` is set before calling it.

File: tests/Utilities/Plugin_Request_Utility_Tests.php [hunk 2]
```diff
@@ -57,4 +57,78 @@ public function test_initialize_runner_with_ajax() {
...
+		// Determine if the cleanup function was run.
+		$cleanup = ! has_filter( 'option_active_plugins' );
+		$runner  = Plugin_Request_Utility::get_runner();
+
+		$wpdb->set_prefix( $table_prefix );
+
+		$this->assertTrue( $prepared );
+		$this->assertTrue( $cleanup );
+		$this->assertNull( $runner );
+	}
```
Comment (felixarntz) [inline tests/Utilities/Plugin_Request_Utility_Tests.php:133]: Based on my feedback, would be good to add one more test here where the test simply calls the `destroy_runner` method to ensure that also works when no runner inside the class was initialized.

**Review by felixarntz — COMMENTED** [review]
@jjgrainger Almost LGTM, just one issue to address.

**Comment by jjgrainger** [comment]
Thanks @felixarntz I've updated the PR with you feedback

**Review by felixarntz — APPROVED** [review]
Thanks @jjgrainger!

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks @jjgrainger, LGTM!

**Pull request merged**